app/scripts/actions/projectActions.js

```diff
@@ -557,37 +558,30 @@ ProjectActions.getChunkUrl.preEmit = function (uploadId, chunkBlob, chunkNum, si
 }
 
 function uploadChunk(uploadId, presignedUrl, chunkBlob, size, parentId, parentKind, chunkNum) {
-    let uploaders = [];
-    let progress = [];
     var xhr = new XMLHttpRequest();
-    uploaders.push(xhr);
     xhr.upload.onprogress = uploadProgress;
     function uploadProgress(e) {
         if (e.lengthComputable) {
             let percentOfChunkUploaded = (e.loaded / e.total) * 100;
-            if(percentOfChunkUploaded < 100){
-                ProjectActions.computeUploadProgress(percentOfChunkUploaded);
-            }
-        } else {
-            alert('cant compute progress')
+            ProjectActions.computeUploadProgress(percentOfChunkUploaded);
```

Comment:
instead of having this action, lets change our existing action to `updateAndProcessChunks(uploadId, chunkNum, chunkUpdates)` where chunkUpdates is an object that optionally has keys `status` and `progress`. Here you would only set the progress, i.e. updateAndProcessChunks(uploadId, chunkNum, {progress: percentOfChunkUploaded})`